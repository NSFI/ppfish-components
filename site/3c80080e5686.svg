<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 24.2.3, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="图层_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 16 16" style="enable-background:new 0 0 16 16;" xml:space="preserve">
<style type="text/css">
	.st0{fill:#fff;}
</style>
<g id="FD-WebDesign">
	<g id="Fish-Design-1.2" transform="translate(-1104.000000, -428.000000)">
		<g id="编组" transform="translate(785.000000, 274.000000)">
			<g id="Group-17备份-2" transform="translate(177.000000, 142.000000)">
				<g id="fd-web-2.3-qr" transform="translate(142.000000, 12.000000)">
					<path id="形状" class="st0" d="M0,15.3C0,15.7,0.3,16,0.7,16h5.1v-5.8H0V15.3L0,15.3z M2.2,12.4h1.5v1.5H2.2V12.4z M7.3,13.1
						h1.5V16H7.3V13.1z M0,7.3h2.9v1.5H0V7.3z M7.3,7.3H4.4v1.5h2.9v2.9h1.5V4.4H7.3V7.3z M10.2,10.2h2.9v2.9h-2.9V10.2z M5.8,0H0.7
						C0.3,0,0,0.3,0,0.7v5.1h5.8V0z M3.7,3.6H2.2V2.2h1.5V3.6z M15.3,0h-5.1v5.8H16V0.7C16,0.3,15.7,0,15.3,0z M13.8,3.6h-1.5V2.2
						h1.5V3.6z M7.3,0h1.5v2.9H7.3V0z M11.6,7.3H16v1.5h-4.4V7.3z"/>
					<path id="路径" class="st0" d="M14.5,14.5h-4.4V16h5.1c0.4,0,0.7-0.3,0.7-0.7v-5.1h-1.5V14.5L14.5,14.5z"/>
				</g>
			</g>
		</g>
	</g>
</g>
</svg>
