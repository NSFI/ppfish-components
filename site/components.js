import React from 'react';
import {BackTop, Row, Col, Menu, Icon, Divider} from 'antd';
import Layout from './layout';
import locales from './locales';
import components from './componentsPage';

const SubMenu = Menu.SubMenu;

export default class Components extends React.Component {
  constructor(props) {
    super(props);
    const menuList = [];
    // 开发指南
    Object.keys(components.documents).map(key => {
      menuList.push({
        key: this.getLocale(`page.${key}`),
        url: `#/components/${key}`
      });
    });
    // 基础组件
    Object.keys(components.list).map((group) => (
      Object.keys(components.list[group]).map(key => {
        menuList.push({
          key: this.getLocale(`page.${key}`),
          url: `#/components/${key}`
        });
      })
    ));
    this.state = {menuList};
  }

  componentWillMount() {
    window.addEventListener("hashchange", this.setComponentShow, false);
  }

  componentDidMount() {
    this.setPage();
  }

  componentWillUnmount() {
    window.removeEventListener("hashchange", this.setComponentShow, false);
  }

  setComponentShow = () => {
    window.scrollTo(0, 0);

    this.setPage();
  };

  getLocale(key) {
    const map = locales['zh-CN'] || {};
    return key.split('.').reduce((a, b) => {
      const parent = map[a];

      if (b) {
        return (parent || {})[b];
      }

      return parent;
    });
  }

  getPage() {
    const routes = location.hash.match(/(?:\/(.+))?\/(.+)/);

    if (routes) {
      return routes[2];
    }

    return 'quick-start';
  }

  setPage(fn) {
    this.setState({page: this.getPage()}, fn);
  }

  //根据page参数获取对应的页的markdown文件并解析
  getComponent(page) {
    this.components = this.components || Object.assign(Object.values(components.list).reduce((a, b) => {
      return Object.assign(a, b);
    }, {}), components.documents);

    const result = this.components[page] || this.components['quick-start'];
    if (result) {
      return React.createElement(result, {
        locale: {
          show: this.getLocale('markdown.show'),
          hide: this.getLocale('markdown.hide')
        }
      });
    }
  }

  render() {
    const componentIndex = this.state.menuList.findIndex((menuItem) => menuItem.key === this.getLocale(`page.${this.state.page}`));
    const lastLink = this.state.menuList[componentIndex - 1];
    const nextLink = this.state.menuList[componentIndex + 1];
    const Navigation = (
      <div>
        <Divider/>
        <Row className="u-navigation-btm">
          <Col span={12} className="prev-page">
            {lastLink && <a href={lastLink.url}><Icon type="arrow-left" className="prev-page-icon"/>{lastLink.key}</a>}
          </Col>
          <Col span={12} className="next-page">
            {nextLink && <a href={nextLink.url}>{nextLink.key}<Icon type="arrow-right" className="next-page-icon"/></a>}
          </Col>
        </Row>
      </div>
    );
    return (
      <Layout className="doc">
        <Row>
          <Col xs={24} sm={24} md={24} lg={6} xl={5} xxl={4}>
            <nav className="side-nav">
              <Menu
                selectedKeys={[this.state.page]}
                defaultOpenKeys={[this.getLocale('misc.development'), this.getLocale('misc.components')]}
                mode="inline"
              >
                <SubMenu key={this.getLocale('misc.development')} title={this.getLocale('misc.development')}>
                  {
                    Object.keys(components.documents).map(page => {
                      return (
                        <Menu.Item key={page}>
                          <a href={`#/components/${page}`}>{this.getLocale(`page.${page}`)}
                          </a>
                        </Menu.Item>
                      );
                    })
                  }
                </SubMenu>
                <SubMenu key={this.getLocale('misc.components')} title={this.getLocale('misc.components')}>
                  {
                    Object.keys(components.list).map(group => {
                      return (
                        <Menu.ItemGroup key={group} title={group} disabled={false}>
                          {
                            Object.keys(components.list[group]).map(page => {
                              return (
                                <Menu.Item key={page}>
                                  <a href={`#/components/${page}`}>{this.getLocale(`page.${page}`)}</a>
                                </Menu.Item>
                              );
                            })
                          }
                        </Menu.ItemGroup>
                      );
                    })
                  }
                </SubMenu>
              </Menu>
            </nav>
          </Col>
          <Col xs={24} sm={24} md={24} lg={18} xl={19} xxl={20}>
            <div className="content">
              <article className="markdown">
                {this.getComponent(this.state.page)}
                {Navigation}
                <BackTop/>
              </article>
            </div>
          </Col>
        </Row>
      </Layout>
    );
  }
}
