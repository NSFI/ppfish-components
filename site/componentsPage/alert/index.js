import Markdown from '../../../libs/markdown';

export default class Alert extends Markdown {
  document(locale) {
    return require(`../../docs/${locale}/alert.md`);
  }
}
