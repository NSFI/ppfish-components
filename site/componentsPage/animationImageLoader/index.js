import Markdown from '../../../libs/markdown';

export default class AnimationImageLoader extends Markdown {
  document(locale) {
    return require(`../../docs/${locale}/animationImageLoader.md`);
  }
}
