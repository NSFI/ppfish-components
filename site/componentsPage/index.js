/*
类型一
    'demo1': {
      type: 'markdown',
      name: 'name1',
      style:require('...')
    },
类型二
    'demo2': {
      type: 'react',
      name: 'name2',
      style:require('...'),
      component:require('...')
    },
*/
export default {
  documents: {
    'quickStart': {
      type: 'markdown',
      name: '快速上手',
    },
    'changelog': {
      type: 'markdown',
      name: '更新日志',
    },
    'contributing': {
      type: 'markdown',
      name: '贡献指南',
    },
    'resource': {
      type: 'markdown',
      name: '设计资源',
    },
    'customTheme': {
      type: 'markdown',
      name: '定制主题',
    },
  },
  list: {
    '通用组件 General': {
      'alert': {
        type: 'markdown',
        name: 'Alert 警告提示',
      },
      'audioPlayer': {
        type: 'markdown',
        name: 'AudioPlayer 音频播放器',
      },
      'avatar': {
        type: 'markdown',
        name: 'Avatar 头像',
      },
      'badge': {
        type: 'markdown',
        name: 'Badge 徽标数',
      },
      'breadcrumb': {
        type: 'markdown',
        name: 'Breadcrumb 面包屑',
      },
      'button': {
        type: 'markdown',
        name: 'Button 按钮',
      },
      'calendar': {
        type: 'markdown',
        name: 'Calendar 日历',
      },
      'card': {
        type: 'markdown',
        name: 'Card 卡片',
      },
      'carousel': {
        type: 'markdown',
        name: 'Carousel 走马灯',
      },
      'cascader': {
        type: 'markdown',
        name: 'Cascader 级联选择',
      },
      'colorPicker': {
        type: 'markdown',
        name: 'ColorPicker 颜色选择器',
      },
      'checkbox': {
        type: 'markdown',
        name: 'Checkbox 多选框',
      },
      'collapse': {
        type: 'markdown',
        name: 'Collapse 折叠面板',
      },
      'datePicker': {
        type: 'markdown',
        name: 'DatePicker 日期选择框',
      },
      'divider': {
        type: 'markdown',
        name: 'Divider 分隔符',
      },
      'drawer': {
        type: 'markdown',
        name: 'Drawer 抽屉组件',
      },
      'dropdown': {
        type: 'markdown',
        name: 'Dropdown 下拉菜单',
      },
      'form': {
        type: 'markdown',
        name: 'Form 表单',
      },
      'grid': {
        type: 'markdown',
        name: 'Grid 栅格系统',
      },
      'imageLoader': {
        type: 'markdown',
        name: 'ImageLoader 图片加载器',
      },
      'input': {
        type: 'markdown',
        name: 'Input 输入框',
      },
      'layout': {
        type: 'markdown',
        name: 'Layout 布局模式',
      },
      'list': {
        type: 'markdown',
        name: 'List 列表',
      },
      'menu': {
        type: 'markdown',
        name: 'Menu 导航菜单',
      },
      'message': {
        type: 'markdown',
        name: 'Message 全局提示',
      },
      'modal': {
        type: 'markdown',
        name: 'Modal 对话框',
      },
      'notification': {
        type: 'markdown',
        name: 'Notification 通知提醒框',
      },
      'pagination': {
        type: 'markdown',
        name: 'Pagination 分页',
      },
      'picturePreview': {
        type: 'markdown',
        name: 'PicturePreview 图片查看器',
        style: require('./picturePreview/index.less')
      },
      'progress': {
        type: 'markdown',
        name: 'Progress 进度条',
      },
      'radio': {
        type: 'markdown',
        name: 'Radio 单选框',
      },
      'richEditor': {
        type: 'markdown',
        name: 'RichEditor 富文本编辑器',
      },
      'select': {
        type: 'markdown',
        name: 'Select 选择器',
      },
      'slider': {
        type: 'markdown',
        name: 'Slider 滑动输入条',
      },
      'spin': {
        type: 'markdown',
        name: 'Spin 加载中',
      },
      'steps': {
        type: 'markdown',
        name: 'Steps 步骤条',
      },
      'switch': {
        type: 'markdown',
        name: 'Switch 开关',
      },
      'table': {
        type: 'markdown',
        name: 'Table 表格',
      },
      'tabs': {
        type: 'markdown',
        name: 'Tabs 标签页',
      },
      'tag': {
        type: 'markdown',
        name: 'Tag 标签',
      },
      'timeline': {
        type: 'markdown',
        name: 'Timeline 时间轴',
      },
      'timePicker': {
        type: 'markdown',
        name: 'TimePicker 时间选择框',
      },
      'tooltip': {
        type: 'markdown',
        name: 'Tooltip 文字提示',
      },
      'transfer': {
        type: 'markdown',
        name: 'Transfer 穿梭框',
      },
      'tree': {
        type: 'markdown',
        name: 'Tree 树形控件',
      },
      'treeSelect': {
        type: 'markdown',
        name: 'TreeSelect 树选择',
      },
      'upload': {
        type: 'markdown',
        name: 'Upload 上传',
      },
      'videoViewer': {
        type: 'markdown',
        name: 'VideoViewer 视频查看器',
      }
    },
    '业务组件 Business': {
      'bizLoading': {
        type: 'markdown',
        name: 'BizLoading 加载动画',
      },
      'bizTable': {
        type: 'markdown',
        name: 'BizTable 自定义表格',
      },
      'bizTimePicker': {
        type: 'markdown',
        name: 'BizTimePicker 自定义时间选择器',
      },
    }
  },
  patterns: {
    '设计要素 Factor': {
      'ruleColor': {
        type: 'react',
        name: 'Color 色彩',
        component: require('./ruleColor')
      },
      'ruleText': {
        type: 'react',
        name: 'Typography 字体',
        component: require('./ruleText')
      },
    },
    '交互规则 Rule': {
      'ruleNumber': {
        type: 'markdown',
        name: '数字',
      },
      'ruleDatetime': {
        type: 'markdown',
        name: '日期时间',
      },
    }
  }
};
