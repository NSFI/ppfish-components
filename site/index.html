<!DOCTYPE html>
<html>
  <head>
    <title>Fish Design</title>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="shortcut icon" href="site/assets/favicon.ico">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.css" />
  </head>
  <body>
    <div id="app"></div>
  </body>
</html>
