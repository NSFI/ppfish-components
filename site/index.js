import 'core-js';
import React from 'react';
import {render} from 'react-dom';
import {AppContainer} from 'react-hot-loader';
import {Router, Route, hashHistory, Redirect, IndexRoute} from 'react-router';
import Components from './components';
import Home from './home';
import Spec from './spec';
import Demo from './demo';
import './styles/index.less';

render(<AppContainer>
  <Router history={hashHistory}>
    <Route path="/home" component={Home}/>
    <Route path="/spec" component={Spec}/>
    <Route path="/components" component={Components}>
      <IndexRoute component={Demo}/>
      <Route path=":demo" component={Demo}/>
    </Route>
    <Redirect from="*" to="/home"/>
  </Router>
</AppContainer>, document.getElementById('app'));

if (module.hot) {
  module.hot.accept('./components', () => {
    const Components = require('./components').default;
    render(<AppContainer>
      <Router history={hashHistory}>
        <Route path="/home" component={Home}/>
        <Route path="/spec" component={Spec}/>
        <Route path="/components" component={Components}>
          <IndexRoute component={Demo}/>
          <Route path=":demo" component={Demo}/>
        </Route>
        <Redirect from="*" to="/home"/>
      </Router>
    </AppContainer>, document.getElementById('app'));
  });
}
