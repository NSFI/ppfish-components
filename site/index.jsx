import React from 'react';
import { render } from 'react-dom';
import { AppContainer } from 'react-hot-loader';

import 'core-js';

import 'element-theme-default';

import './styles/base.scss';
import './styles/prism.css';

import App from './page';

render(<AppContainer><App /></AppContainer>, document.getElementById('app'));

if (module.hot) {
  module.hot.accept('./page', () => {
    const App = require('./page').default;

    render(<AppContainer><App /></AppContainer>, document.getElementById('app'));
  });
}
