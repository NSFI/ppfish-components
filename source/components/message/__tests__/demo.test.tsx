import {plainComponents} from '../../../../site/componentsPage';

let demo_component = 'message';

let demoName = demo_component.substr(0, 1).toLowerCase() + demo_component.substr(1);
const menuItem = plainComponents.find(item => item.key === demoName);

if(menuItem){
  const demoTest = require('../../../../tools/tests/demoTest').default;
  demoTest(demoName);
}else{
  xit('skip',()=>{});
}
