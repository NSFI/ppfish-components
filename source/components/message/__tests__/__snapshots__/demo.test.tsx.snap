// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders message demo correctly 1`] = `
<div>
  <button
    class="fishd-btn fishd-btn-primary"
    type="button"
  >
    <span>
      Display normal message
    </span>
  </button>
  <div
    class="components-message-demo-other"
  >
    <button
      class="fishd-btn"
      type="button"
    >
      <span>
        Success
      </span>
    </button>
    <button
      class="fishd-btn"
      type="button"
    >
      <span>
        Error
      </span>
    </button>
    <button
      class="fishd-btn"
      type="button"
    >
      <span>
        Warning
      </span>
    </button>
  </div>
  <button
    class="fishd-btn"
    type="button"
  >
    <span>
      Customized display duration
    </span>
  </button>
  <button
    class="fishd-btn"
    type="button"
  >
    <span>
      Display a loading indicator
    </span>
  </button>
  <button
    class="fishd-btn"
    type="button"
  >
    <span>
      Display a sequence of message
    </span>
  </button>
</div>
`;
